charts/minio-manager/templates/grant-access-job.yaml

```diff
@@ -0,0 +1,41 @@
+{{- range $.Values.users }}
+{{- $policy := ( .policy | default "readonly") -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: grant-access-{{ .passwordSecretName }}-{{ $policy }}
+spec:
+  backoffLimit: 4
+  activeDeadlineSeconds: 120
+  template:
+    metadata:
+      {{- with $.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      containers:
+      - name: grant-access
+        image: "{{ $.Values.image.repository }}:{{ $.Values.image.tag }}"
+        imagePullPolicy: {{ $.Values.image.pullPolicy }}
+        command:
+          - /bin/sh
+          - -c
+          - -e
+          - |
+            while [ -z "$(mc config host add minio ${MINIO_ENDPOINT} ${MINIO_ACCESS_KEY} ${MINIO_SECRET_KEY})" ] ; do echo 'Wait minio to startup...' && sleep 0.1; done;
+            while [ -z "$(mc admin user info minio ${USER})" ] ; do echo 'Wait user ${USER} to be created...' && sleep 0.1; done;
```

Comment:
I looked at another pattern another day to manage these dependencies, what you can do here is have just one Job and add the create_buckets and create_roles containers as init containers of create_users container. That will also make things clearer in case of failures.

Reply:
That's a good idea the init container on a job.
And for the moment I'm not creating the buckets, but good idea to add that too. 👍